File: 2024/May/5-8 506 Relative Ranks/RelativeRanks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        HashMap<Integer, Integer> indexes = new HashMap<Integer, Integer>();
        String[] answer = new String[score.length];

        for(int i = 0; i < score.length; i++) {
            indexes.put(score[i], i);
        }

        Arrays.sort(score);

        int place = 1;
        for(int i = score.length - 1; i >= 0; i--) {
            String str = "";
            switch(place) {
                case 1:
                    str = "Gold Medal";
                    break;
                case 2:
                    str = "Silver Medal";
                    break;
                case 3:
                    str = "Bronze Medal";
                    break;
                default:
                    str = String.valueOf(place);
                    break;
            }
        
            answer[indexes.get(score[i])] = str;
            place++;
        }


        return answer;
    }
}

File: 2024/May/5-8 506 Relative Ranks/RelativeRanks.js
/**
 * @param {number[]} score
 * @return {string[]}
 */
var findRelativeRanks = function(score) {
    let indexes = new Map();
    let answer = new Array(score.length);

    for(let i = 0; i < score.length; i++) {
        indexes.set(score[i], i);
    }

    score.sort((a, b) => b - a);
    let place = 1;
    for(let s of score) {
        let str = '';
        switch(place) {
            case 1:
                str = 'Gold Medal';
                break;
            case 2:
                str = 'Silver Medal';
                break;
            case 3:
                str = 'Bronze Medal';
                break;
            default:
                str = String(place);
                break;
        }

        answer[indexes.get(s)] = str;
        place++;
    }

    return answer;
};